src/rust/plugin-work-queue/src/server.rs

```diff
@@ -24,112 +31,292 @@ use rust_proto::plugin_work_queue::{
+
+        Ok(PutExecuteAnalyzerResponse {})
     }
 
     #[allow(dead_code)]
     async fn get_execute_generator(
         &self,
         _request: GetExecuteGeneratorRequest,
     ) -> Result<GetExecuteGeneratorResponse, PluginWorkQueueError> {
-        todo!()
+        let message = self.queue.get_generator_message().await?;
+        let message = match message {
+            Some(message) => message,
+            None => {
+                return Ok(GetExecuteGeneratorResponse {
+                    execution_job: None,
+                    request_id: 0,
+                })
+            }
+        };
+        let execution_job = ExecutionJob {
+            tenant_id: message.request.tenant_id,
+            plugin_id: message.request.plugin_id,
+            data: message.request.pipeline_message,
+        };
+        Ok(GetExecuteGeneratorResponse {
+            execution_job: Some(execution_job),
+            request_id: message.request.execution_key.into(),
+        })
     }
 
     #[allow(dead_code)]
     async fn get_execute_analyzer(
         &self,
         _request: GetExecuteAnalyzerRequest,
     ) -> Result<GetExecuteAnalyzerResponse, PluginWorkQueueError> {
-        todo!()
+        let message = self.queue.get_analyzer_message().await?;
+        let message = match message {
+            Some(message) => message,
+            None => {
+                return Ok(GetExecuteAnalyzerResponse {
+                    execution_job: None,
+                    request_id: 0,
+                })
+            }
+        };
+        let execution_job = ExecutionJob {
+            tenant_id: message.request.tenant_id,
+            plugin_id: message.request.plugin_id,
+            data: message.request.pipeline_message,
+        };
+        Ok(GetExecuteAnalyzerResponse {
+            execution_job: Some(execution_job),
+            request_id: message.request.execution_key.into(),
+        })
     }
 
     #[allow(dead_code)]
-    async fn acknowledge(
+    async fn acknowledge_generator(
         &self,
-        _request: AcknowledgeRequest,
-    ) -> Result<AcknowledgeResponse, PluginWorkQueueError> {
-        todo!()
+        request: AcknowledgeGeneratorRequest,
+    ) -> Result<AcknowledgeGeneratorResponse, PluginWorkQueueError> {
+        let status = match request.success {
+            true => crate::psql_queue::Status::Processed,
+            false => crate::psql_queue::Status::Failed,
+        };
+        self.queue
+            .ack_generator(request.request_id.into(), status)
+            .await?;
+        Ok(AcknowledgeGeneratorResponse {})
+    }
+
+    #[allow(dead_code)]
+    async fn acknowledge_analyzer(
+        &self,
+        request: AcknowledgeAnalyzerRequest,
+    ) -> Result<AcknowledgeAnalyzerResponse, PluginWorkQueueError> {
+        let status = match request.success {
+            true => crate::psql_queue::Status::Processed,
+            false => crate::psql_queue::Status::Failed,
+        };
```

Comment:
I could do that, but it would mean that `Status` is a type that `rust-proto` would be aware of, as opposed to being specific to the representation in the queue. That's not so trivial - it also means sqlx becomes a dependency of rust-proto, which maybe is fine, but I think merits some discussion about where these sorts of types live.

Instead I can add a From<bool> to Status, which keeps things separate.

I can add  the use.

Reply:
On second thought I don't think that a From<bool> would be proper. It's too general. I think it's best to leave the two conversions in place for now and address this in later PRs.